### PR TITLE
More Outbox details, DTC to Outbox conversion

### PR DESCRIPTION
**Pull request by DavidBoike**
Trying to explain the Outbox better and provide a path for how to move a system from DTC to Outbox.

Please review @Particular/nservicebus-maintainers @Particular/docs-maintainers

### CODE REVIEW

File: nservicebus/outbox/index.md [hunk 1]
```diff
@@ -41,7 +45,7 @@ Note: On the wire level the Outbox guarantees `at-least-once` message delivery,
 
 ## Important design considerations
 
- * The business data and deduplication data need to be stored in the same database.
+ * Business data and Outbox deduplication data must be stored in the same database, in order to take advantage of the same local database transaction to store all data.
```
Comment (DavidBoike) [inline nservicebus/outbox/index.md]: I feel like this is the single biggest caveat to Outbox and the biggest thing customers miss. I'm tempted to make it bold.

Comment (bording) [inline nservicebus/outbox/index.md]: I would make this a warning then. 

Also, pretty sure that sentence doesn't need a comma!

File: nservicebus/outbox/index.md [hunk 2]
```diff
@@ -56,6 +60,19 @@ partial: enable-outbox
 To learn about Outbox configuration options such as time to keep deduplication data or deduplication data clean up interval, refer to the dedicated pages for [NHibernate](/persistence/nhibernate/outbox.md), [RavenDB](/persistence/ravendb/outbox.md) or [ServiceFabric](/persistence/service-fabric/outbox.md) persistence.
 
 
+## Converting from DTC to Outbox
+
+When converting a system from using the DTC to the Outbox, care must be taken to ensure the system does not process duplicate messages incorrectly.
+
+Because the Outbox feature uses an "at least once" consistency guarantee at the transport level, endpoints that enable the Outbox will occasionally send duplicate messages. These duplicate messages will be properly handled by deduplication by other Outbox-enabled endpoints, but would be processed in duplicate by endpoints still using the DTC.
```
Comment (bording) [inline nservicebus/outbox/index.md]: Too many "bys" in the second sentence.

File: nservicebus/outbox/index.md [hunk 3]
```diff
@@ -15,7 +15,11 @@ related:
...
+The Outbox is an infrastructure feature designed to simulate the reliability of distributed transactions (using the Distributed Transactions Coordinator or DTC) while not actually using the DTC.
+
+The purpose of the DTC is to guarantee consistency between messaging operations and the data persistence. Messaging operations includes the message being processed as well as any messages being sent out as a result. Data persistence includes any business data persisted by a message handler, as well as any NServiceBus saga or timeout data stored at the same time. The DTC ensures that all these operations either complete successfully together, or are all rolled back.
+
+Instead of the DTC, the NServiceBus Outbox feature can be used to mimic this level of consistency without the need for distributed transactions. It does this by first storing any outgoing messages in the database, taking advantage of the same (non-distributed) local transaction to ensure that the messaging operations are stored atomically with any changes to business data and NServiceBus data. Once that transaction is successfully committed, the stored outgoing messages are dispatched to the message transport separately.
```
Comment (bording) [inline nservicebus/outbox/index.md]: The phrase "message transport" seems off. I don't think we generally refer to transports that way elsewhere. Perhaps "dispatched to their destinations", or maybe just "dispatched."

File: nservicebus/outbox/index.md (hunk 1 above)
Comment (SimonCropp) [inline nservicebus/outbox/index.md]: happy with a warning. but i would invert the text

> To take advantage of the same local database transaction to store all data, and get a similar experience to using DTC, the business data and Outbox deduplication data must be stored in the same database.

**Pull request merged**